ArduCopter/RC_Channel.cpp

```diff
@@ -685,6 +691,29 @@ void RC_Channel_Copter::do_aux_function_change_force_flying(const AuxSwitchPos c
     }
 }
 
+#if QUICKTUNE_ENABLED
+// called on any Quicktune Aux function change
+void RC_Channel_Copter::do_aux_function_quicktune(const AuxSwitchPos ch_flag)
+{
+    if (copter.quicktune == nullptr && !copter.arming.is_armed()) {
+        //first call, allocate quicktune object
+        GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Quicktune: Initialised.");
+        copter.quicktune = NEW_NOTHROW AP_Quicktune();
+        if (copter.quicktune == nullptr) {
+            // Can't use BoardConfig error as this might happen while flying.
+            GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "Quicktune: unable to allocate.");
+            return;
+        }
+        AP_Param::load_object_from_eeprom(copter.quicktune, copter.quicktune->var_info);
```

Comment:
Why don't we do the allocation if the aux switch is setup. It will work for those with physical switches.

Reply:
It gets allocated anyway - if the RC channel option is set, it gets allocated on boot.